src/domain/RestaurantData.ts

```diff
@@ -0,0 +1,83 @@
+import ERROR_MASSAGE from "../constants/errorMessage.js";
+import Category from "../types/Category.js";
+import Restaurant from "../types/Restaurant.js";
+
+const CATEGORY_IMAGE: Readonly<Record<Category, string>> = Object.freeze({
+  한식: "./category-korean.png",
+  중식: "./category-chinese.png",
+  일식: "./category-japanese.png",
+  양식: "./category-western.png",
+  아시안: "./category-asian.png",
+  기타: "./category-etc.png",
+});
+
+export default class RestaurantData {
+  id: string;
+  src: string;
+  alt: string;
+  name: string;
+  distance: number;
+  description: string;
+  link: string;
+  category: Category;
+  isWish: boolean;
+
+  constructor({ id, name, distance, description = "", link = "", category, isWish }: Restaurant) {
+    this.validateCategory(category);
+    this.validateDistance(distance);
+    this.validateName(name);
+
+    this.id = id || crypto.randomUUID();
+    this.src = CATEGORY_IMAGE[category];
+    this.alt = category;
+    this.name = name;
+    this.distance = Number(distance);
+    this.description = description;
+    this.link = link;
+    this.category = category;
+    this.isWish = isWish;
+  }
+
+  getData(): Restaurant {
+    return {
+      id: this.id,
+      src: this.src,
+      alt: this.alt,
+      name: this.name,
+      distance: this.distance,
+      description: this.description,
+      link: this.link,
+      category: this.category,
+      isWish: this.isWish
+    };
+  }
+
+  getId(): string {
+    return this.id;
+  }
+
+  toggleIsWish(): void {
+    this.isWish = !this.isWish;
+  }
+
+  isValidateOption(value: string | number): boolean {
+    return !value;
+  }
+
+  isValidateName(name: string): boolean {
+    const NAME_LENGTH_MIN = 2;
+    return name.length < NAME_LENGTH_MIN;
+  }
+
+  validateCategory(category: string): void {
+    if (this.isValidateOption(category)) throw Error(ERROR_MASSAGE.category);
+  }
+
+  validateDistance(distance: number): void {
+    if (this.isValidateOption(distance)) throw Error(ERROR_MASSAGE.distance);
+  }
```

Comment:
메서드를 사용하는 것보다 아래처럼 코드가 작성되어 있는게 더 직관적으로 코드를 이해할 수 있겠다는 생각이 드는데요. 메서드로 만드신 이유가 있을까요?

```suggestion
  validateDistance(distance: number): void {
    if (!distance) throw Error(ERROR_MASSAGE.distance);
  }
```

Reply:
추가로 distance가 `0`인 경우에도 에러를 뱉게 될텐데 요건 의도하신 부분일까용?

Reply:
1) `!distance` 코드가 더 직관적이지만 함수로 만든 이유는 단일 책임 원칙을 따랐기 때문이에요.

`검증하기`와 `검증 결과에 따라 행동하기`는 각각 별개의 일이라고 생각됩니다. 그래서 코드가 짧더라도 여러 가지 일을 하지 않도록 분리했어요.

Reply:
2. 아니요, 0일 때는 고려하지 않았는데, 그 부분까지 걸러지는 상황이 발생했네요. 🫣
지금처럼 `!` 연산자를 쓸 때 정상 값인 0을 비정상 값으로 만드는 상황을 조심해야겠어요.

9f0869d1ada83afdefaf81c8378428d1d9aa92f7

Reply:
>유조는 가독성 측면에서 메서드로 따로 분리하지 않는 게 더 나을 때도 있다고 생각하시는 게 맞을까요?

JavaScript의 기본 문법이나 내장 메서드를 사용한 동작만 수행하고 있다면 메서드로 분리하지 않는 것을 선호해요

메서드가 어떤 동작을 하는지는 모르지만 기본 문법이나 내장 메서드는 JavaScript 코드를 작성하는 모두가 알고 있는 내용이기 때문에 코드를 이해하기 위해 추가적인 탐색이 필요 없기 때문인데요.

이런 생각을 가진 사람이 있다 정도로만 알고 넘어가셔도 될거 같아요 ㅎㅎ